pytroll_collectors/s3stalker.py

```diff
@@ -88,23 +111,26 @@ def _match_files_to_pattern(files, path, pattern):
     return files
 
 
-def set_last_fetch(timestamp):
-    """Set the last fetch time."""
-    DatetimeHolder.last_fetch = timestamp
-
-
-def publish_new_files(bucket, config):
+def publish_new_files(bucket, config, publisher_ready_time=2.5):
     """Publish files newly arrived in bucket."""
+    time_back = config.pop('fetch_back_to')
+    set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
     with Publish("s3_stalker") as pub:
-        time_back = config['timedelta']
-        subject = config['subject']
-        pattern = config.get('file_pattern')
-        with sleeper(2.5):
-            set_last_fetch(datetime.now(tz.UTC) - timedelta(**time_back))
-            s3_kwargs = config['s3_kwargs']
-            fs, files = get_last_files(bucket, pattern=pattern, **s3_kwargs)
-            messages = filelist_unzip_to_messages(fs, files, subject)
-
+        with sleeper(publisher_ready_time):
+            messages = create_messages_for_recent_files(bucket, config)
         for message in messages:
             logger.info("Publishing %s", str(message))
             pub.send(str(message))
+
+
+def create_messages_for_recent_files(bucket, config):
+    """Create messages for recent files and return."""
+    logger.debug("Create messages for recent files...")
+
+    logger.debug(f"Last fetch is {get_last_fetch()}")
```

Comment:
This message is logged also when the file list is retrieved, one of them should be removed.

Reply:
done